src/Adapter/S3Adapter.php

```diff
@@ -106,35 +108,53 @@ public function setCannedAcl(string $cannedAcl)
 
     public function setData($key, $data)
     {
-        $this->s3client->putObject(
-            [
-                'Bucket' => $this->bucketname,
-                'Key' => $this->prefix . $key,
-                'Body' => $data,
-                'ACL' => $this->cannedAcl,
-            ]
-        );
+        try {
+            $this->s3client->putObject(
+                [
+                    'Bucket' => $this->bucketname,
+                    'Key' => $this->prefix . $key,
+                    'Body' => $data,
+                    'ACL' => $this->cannedAcl,
+                ]
+            );
+        } catch (S3Exception $e) {
+            throw new AdapterException('Failed to store data because of an S3 error', null, $e);
+        } catch (AwsException $e) {
+            throw new AdapterException('Failed to store data because of an AWS error', null, $e);
+        }
```

Comment:
So all this change really does is catch AWS and S3 exceptions and throw AdapterException instead.